src/Symfony/Component/DependencyInjection/Tests/Fixtures/includes/autowiring_classes.php

```diff
@@ -157,7 +157,7 @@ public function __construct(Dunglas $j, Dunglas $k)
 
 class OptionalParameter
 {
-    public function __construct(CollisionInterface $c = null, A $a, Foo $f = null)
+    public function __construct(A $a, ?CollisionInterface $c = null, ?Foo $f = null)
```

Comment:
Swapped because it triggers a deprecation : optional parameter cannot be before mandatory parameter.

Reply:
this test case was about optional argument being before mandatory ones, so we'd better keep this and mark `testOptionalParameter` with `@group legacy`

Reply:
Alright, added. However, I was not able to catch the deprecation with the `ExpectDeprecationTrait`, thus leading to have it in the test report. Is there another way to expect it, as it is raised by the language?

Reply:
Ah right, this should be moved to a standalone fixture file

Reply:
Here you go 👍